ecl/hqlcpp/hqlhtcpp.cpp

```diff
@@ -17800,6 +17800,31 @@ void HqlCppTranslator::doBuildAssignRegexFindReplace(BuildCtx & ctx, const CHqlB
     doBuildNewRegexFindReplace(ctx, &target, expr, NULL);
 }
 
+void HqlCppTranslator::doBuildExprRegexFindSet(BuildCtx & ctx, IHqlExpression * expr, CHqlBoundExpr & bound)
+{
+    CHqlBoundExpr boundMatch;
+    if (ctx.getMatchExpr(expr, boundMatch))
+    {
+        bound.set(boundMatch);
+        return;
+    }
+
+    IHqlExpression * pattern = expr->queryChild(0);
+    IHqlExpression * search = expr->queryChild(1);
+    bool isUnicode = isUnicodeType(search->queryType());
+    IHqlExpression * compiled = doBuildRegexCompileInstance(ctx, pattern, isUnicode, !expr->hasAttribute(noCaseAtom));
+
+    // Because the search instance is created locally, the search parameter is always going to be valid
+    // as long as the find instance.  Only exception could be if call created a temporary class instance.
+    bool needToKeepSearchString = true;
```

Comment:
findreplace uses a different class... The replace function has full input string is provided in the arguments.

The needToKeepSearchString forces the string to be stored within the class.  Please see getMatchSet() of CUStrRegExprFindInstance (Line 5113 of eclrtl.cpp) to see how the 'sample' field is used in the search.  Without the flag, the searched string will not be stored locally (in 'sample') and getMatchSet would not be able to parse the entire string just the first one.